resolver.go

```diff
@@ -145,6 +145,14 @@ func (r *resolver) Start() error {
 		return fmt.Errorf("setting up IP table rules failed: %v", err)
 	}
 
+	// If the container goes down quickly after start resolver.Stop() can
+	// get called before Start() completes. Check if the socket is still
+	// valid before Activating the DNS server. This avoids a panic in dns
+	// libarary code because of missing nil check for interface's data.
+	// github.com/docker/docker/issues/28112
+	if r.conn == nil {
```

Comment:
> Its not possible to have resolver.conn != nil && resolver.tcpListen == nil

Yes it is, the stop() may be called after the start() started the udp listener but before it started the tcp listener.

>  Instead I am going to push a PR in miekg/dns code to check the interface datatype being nil

I do not think it is the library responsibility if we `nil` the net.PacketConn after initializing its server structure.

>  I prefer not to add more locks in our code

I am not sure I understand the opposition versus locks when we have the same structure being accessed/modified by different threads.